lib/Doctrine/ORM/Tools/DoctrineSetup.php

```diff
@@ -0,0 +1,199 @@
+use Doctrine\ORM\Mapping\Driver\AttributeDriver;
+use Doctrine\ORM\Mapping\Driver\XmlDriver;
+use Doctrine\ORM\Mapping\Driver\YamlDriver;
+use Memcached;
+use Psr\Cache\CacheItemPoolInterface;
+use Redis;
+use RuntimeException;
+use Symfony\Component\Cache\Adapter\ApcuAdapter;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+use Symfony\Component\Cache\Adapter\MemcachedAdapter;
+use Symfony\Component\Cache\Adapter\RedisAdapter;
+
+use function class_exists;
+use function extension_loaded;
+use function md5;
+use function sys_get_temp_dir;
+
+final class DoctrineSetup
+{
+    /**
+     * Creates a configuration with an annotation metadata driver.
+     *
+     * @param string[] $paths
+     */
+    public static function createAnnotationMetadataConfiguration(
+        array $paths,
+        bool $isDevMode = false,
+        ?string $proxyDir = null,
+        ?CacheItemPoolInterface $cache = null
+    ): Configuration {
+        $config = self::createConfiguration($isDevMode, $proxyDir, $cache);
+        $config->setMetadataDriverImpl(self::createDefaultAnnotationDriver($paths));
+
+        return $config;
+    }
+
+    /**
+     * Adds a new default annotation driver with a correctly configured annotation reader.
+     *
+     * @param string[] $paths
+     */
+    public static function createDefaultAnnotationDriver(
+        array $paths = [],
+        ?CacheItemPoolInterface $cache = null
+    ): AnnotationDriver {
+        $reader = new AnnotationReader();
+
+        if ($cache === null && class_exists(ArrayAdapter::class)) {
+            $cache = new ArrayAdapter();
+        }
+
+        if ($cache !== null) {
+            $reader = new PsrCachedReader($reader, $cache);
+        }
+
+        return new AnnotationDriver($reader, $paths);
+    }
+
+    /**
+     * Creates a configuration with an attribute metadata driver.
+     *
+     * @param string[] $paths
+     */
+    public static function createAttributeMetadataConfiguration(
+        array $paths,
+        bool $isDevMode = false,
+        ?string $proxyDir = null,
+        ?CacheItemPoolInterface $cache = null
+    ): Configuration {
+        $config = self::createConfiguration($isDevMode, $proxyDir, $cache);
+        $config->setMetadataDriverImpl(new AttributeDriver($paths));
+
+        return $config;
+    }
+
+    /**
+     * Creates a configuration with a xml metadata driver.
+     *
+     * @param string[] $paths
+     */
+    public static function createXMLMetadataConfiguration(
+        array $paths,
+        bool $isDevMode = false,
+        ?string $proxyDir = null,
+        ?CacheItemPoolInterface $cache = null
+    ): Configuration {
+        $config = self::createConfiguration($isDevMode, $proxyDir, $cache);
+        $config->setMetadataDriverImpl(new XmlDriver($paths));
+
+        return $config;
+    }
+
+    /**
+     * Creates a configuration with a yaml metadata driver.
+     *
+     * @deprecated YAML metadata mapping is deprecated and will be removed in 3.0
```

Comment:
It felt a bit odd to introduce a method and instantly deprecate it. I decided to add the method to allow switching to a PSR-6 cache before migrating the whole YAML mapping configuration to XML or annotations.

Reply:
This reminds me that we should think about deprecating annotations as well… but maybe it's too early?